src/Elasticsearch/Endpoints/AbstractEndpoint.php

```diff
@@ -60,14 +64,18 @@
     /**
      * @param Transport $transport
      */
-    public function __construct($transport)
+    public function __construct($transport, RequestFactory $requestFactory = null)
     {
         $this->transport = $transport;
+        $this->requestFactory = $requestFactory;
     }
 
     /**
      * @throws \Exception
      * @return array
+     *
+     * @deprecated Performing a request directly on an endpoint is deprecated since 2.2 and will be removed in 3.0,
+     *             you should instead use the createRequest method
      */
     public function performRequest()
```

Comment:
It would be a good proof that it works as expected :-) But lets see what @polyfractal thinks as he is the one that knows side effects / risk etc best.